data/cache_policy.go

```diff
@@ -160,92 +184,26 @@ func (p *CachePolicy) MatchesSizeLimits(size int) bool {
 }
 
 func (p *CachePolicy) EmptyState() common.CacheEmptyBehavior {
-	return p.config.Empty
-}
-
-func (p *CachePolicy) matchParams(params []interface{}) (bool, error) {
-	if len(p.config.Params) == 0 {
-		return true, nil
+	// For legacy configs, use the legacy field
+	if p.config.Empty != 0 || p.config.Network != "" || p.config.Method != "" {
+		return p.config.Empty
 	}
 
-	for i, pattern := range p.config.Params {
-		var v interface{}
-		if i < len(params) {
-			v = params[i]
-		}
-		match, err := matchParam(pattern, v)
-		if err != nil {
-			return false, err
-		}
-		if !match {
-			return false, nil
-		}
-	}
-	return true, nil
-}
-
-func matchParam(pattern interface{}, param interface{}) (bool, error) {
-	switch p := pattern.(type) {
-	case map[string]interface{}:
-		// For objects, recursively match each field
-		paramMap, ok := param.(map[string]interface{})
-		if !ok {
-			return false, nil
-		}
-		for k, v := range p {
-			paramValue, exists := paramMap[k]
-			match, err := matchParam(v, paramValue)
-			if err != nil {
-				return false, err
-			}
-			if !exists || !match {
-				return false, nil
-			}
-		}
-		return true, nil
-	case []interface{}:
-		// For arrays, match each element
-		paramArray, ok := param.([]interface{})
-		if !ok || len(p) != len(paramArray) {
-			return false, nil
-		}
-		for i, v := range p {
-			match, err := matchParam(v, paramArray[i])
-			if err != nil {
-				return false, err
-			}
-			if !match {
-				return false, nil
+	// For new matcher configs, find the first INCLUDE matcher (skip auto-added excludes)
+	if len(p.config.Matchers) > 0 {
+		for _, matcher := range p.config.Matchers {
+			if matcher != nil && matcher.Action == common.MatcherInclude {
+				return matcher.Empty
```

Comment:
Two important aspects for this:
* We should not at need to handle legacy fields here, they must be fully migrated to new approach, and the new approach should be able to behave as we intend, if not possible we'd need a design update.
* The empty state behavior should be used from the exact matcher that matched the request initially, this current code is just covering 1 edge-case which is when we have exclude on default matcher + have different empty on the next matcher.